src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                    }
+                }
+
+                // Recurse for referencing entities
+                foreach (Tuple<string, object?> referencingEntity in multipleCreateStructure.ReferencingEntities)
+                {
+                    string relatedEntityName = GetRelatedEntityNameInRelationship(entity, referencingEntity.Item1);
+                    MultipleCreateStructure referencingEntityMultipleCreateStructure = new(relatedEntityName, entityName, multipleCreateStructure.CurrentEntityPKs, referencingEntity.Item2, IsManyToManyRelationship(entity, referencingEntity.Item1));
+                    IValueNode node = GraphQLUtils.GetFieldNodeForGivenFieldName(parameterNodes, referencingEntity.Item1);
+                    PerformDbInsertOperation(context, node.Value, sqlMetadataProvider, referencingEntityMultipleCreateStructure, primaryKeysOfCreatedItem);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to get the name of the related entity for a given relationship name.
+        /// </summary>
+        /// <param name="entity">Entity object</param>
+        /// <param name="relationshipName">Name of the relationship</param>
+        /// <returns>Name of the related entity</returns>
+        public static string GetRelatedEntityNameInRelationship(Entity entity, string relationshipName)
+        {
+            if (entity.Relationships is null)
+            {
+                throw new DataApiBuilderException(message: "Entity has no relationships defined",
+                                                  statusCode: HttpStatusCode.InternalServerError,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+            }
+
+            if (entity.Relationships.TryGetValue(relationshipName, out EntityRelationship? entityRelationship)
+               && entityRelationship is not null)
+            {
+                return entityRelationship.TargetEntity;
+            }
+            else
+            {
+                throw new DataApiBuilderException(message: $"Entity does not have a relationship named {relationshipName}",
+                                                  statusCode: HttpStatusCode.InternalServerError,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.RelationshipNotFound);
+            }
+
+        }
+
+        /// <summary>
+        /// Helper method to determine whether the relationship is a M:N relationship.
+        /// </summary>
+        /// <param name="entity">Entity </param>
+        /// <param name="relationshipName">Name of the relationship</param>
+        /// <returns>True/False indicating whther a record should be created in the linking table</returns>
+        public static bool IsManyToManyRelationship(Entity entity, string relationshipName)
```

Comment:
A method serving the exact same purpose is present here: https://github.com/Azure/data-api-builder/blob/838af5333934c3cdaf5f1e5031324ad20d1deb78/src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs#L450. Please remove either one of them.

Reply:
Removed this one. Moved `IsMToNRelationship` to `GraphQLUtils` and re-used this in all places.